recbole/model/general_recommender/line.py

```diff
@@ -0,0 +1,182 @@
+# -*- coding: utf-8 -*-
+# @Time   : 2020/12/8
+# @Author : Yihong Guo
+# @Email  : gyihong@hotmail.com
+
+r"""
+LINE
+################################################
+Reference:
+    Jian Tang et al. "LINE: Large-scale Information Network Embedding." in WWW 2015.
+
+Reference code:
+    https://github.com/shenweichen/GraphEmbedding
+"""
+
+import torch
+import torch.nn as nn
+import random
+import numpy as np
+
+from recbole.utils import InputType
+from recbole.model.abstract_recommender import GeneralRecommender
+from recbole.model.init import xavier_normal_initialization
+
+
+class NegSamplingLoss(nn.Module):
+    def __init__(self):
+        super(NegSamplingLoss, self).__init__()
+
+    def forward(self, score, sign):
+        return -torch.mean(torch.sigmoid(sign * score))
+
+
+class LINE(GeneralRecommender):
+    r"""LINE is a graph embedding model.
+
+    We implement the model to train users and items embedding for recommendation.
+    """
+    input_type = InputType.PAIRWISE
+
+    def __init__(self, config, dataset):
+        super(LINE, self).__init__(config, dataset)
+
+        self.embedding_size = config['embedding_size']
+        self.order = config['order']
+        self.second_order_loss_weight = config['second_order_loss_weight']
+        self.training_neg_sample_num = config['training_neg_sample_num']
+
+
+        self.interaction_feat = dataset.dataset.inter_feat
+
+        self.uid_field = dataset.dataset.uid_field
+        self.iid_field = dataset.dataset.iid_field
```

Comment:
There have been the same attributes in the base class ``GeneralRecommender``, please refer to [https://github.com/RUCAIBox/RecBole/blob/511cda89deca404808bea1df2f5e98e17e9015f1/recbole/model/abstract_recommender.py#L82-L84](https://github.com/RUCAIBox/RecBole/blob/511cda89deca404808bea1df2f5e98e17e9015f1/recbole/model/abstract_recommender.py#L82-L84), so there is no need to add these two attributes.

Also, line 75-76 need to be changed.